server/repository/src/db_diesel/patient.rs

```diff
@@ -208,6 +210,20 @@ impl<'a> PatientRepository<'a> {
                 query = query.or_filter(name_dsl::id.eq_any(sub_query))
             }
 
+            if program_enrolment_name.is_some() {
+                let sub_query = ProgramEnrolmentRepository::create_filtered_query(Some(
+                    ProgramEnrolmentFilter {
+                        program_name: program_enrolment_name,
+                        program_context_id: allowed_ctx
+                            .map(|ctxs| EqualFilter::default().restrict_results(ctxs)),
+                        ..Default::default()
+                    },
+                ))
+                .select(program_enrolment_dsl::patient_id);
+
+                query = query.or_filter(name_dsl::id.eq_any(sub_query))
```

Comment:
This should be an AND filter, right? for example, add program id filter and then try to narrow it using the patient id, this doesn work...

Reply:
[992830b](https://github.com/msupply-foundation/open-msupply/pull/2919/commits/992830b3547df6c607497ff5daaafdcda437360e)

Reply:
There is the `name_or_code` OR filter after this filter now. Beside its already not working 100% (was this what I was complaining about before?), could you please move this new filter after the `name_or_code` filter?

Just to show the problem, currently if all filters are set it would generate something like:

```sql
(((((code OR NHI)
   AND name)
     OR code)
       AND program_enrolment_name)
         AND ...)
```

But we want:
```sql
(code OR NHI) AND (name OR code) AND programe_enrolment_name AND ...
```